posthog/queries/query_date_range.py

```diff
@@ -174,21 +174,40 @@ def date_from(self) -> Tuple[str, Dict]:
 
         return date_from_query, date_from_param
 
-    def _get_timezone_aware_date_condition(self, operator: str, date_clause: str) -> str:
-        if self.should_round:
-            return f"AND toDateTime({self._table}timestamp, 'UTC') {operator} {self._timezone_date_clause(date_clause)}"
+    def _get_timezone_aware_date_condition(self, operator: Literal[">=", "<="]) -> str:
+        relevant_param = "date_from" if operator.startswith(">") else "date_to"
+        event_timestamp_expr = self._normalize_datetime(f"{self._table}timestamp", stored_in_utc=True)
+        date_expr = self._normalize_datetime(f"%({relevant_param})s")
+        if operator == ">=" and self.should_round:  # Round date_from to start of interval if `should_round` is true
+            date_expr = self._truncate_normalized_datetime(date_expr, self.interval_annotation)
+        return f"AND {event_timestamp_expr} {operator} {date_expr}"
+
+    @staticmethod
+    def _normalize_datetime(datetime_expr: str, *, stored_in_utc: bool = False) -> str:
+        """Return expression with datetime normalized to project timezone.
+        Use `from_utc=True` for datetimes from the database - we always store datetimes in UTC.
+        """
+        if stored_in_utc:
+            return f"toTimeZone(toDateTime({datetime_expr}), %(timezone)s)"
         else:
-            return (
-                f"AND toDateTime({self._table}timestamp, 'UTC') {operator} toDateTime(%({date_clause})s, %(timezone)s)"
-            )
-
-    def _timezone_date_clause(self, date_clause: str) -> str:
-        clause = f"toDateTime({self.interval_annotation}(toDateTime(%({date_clause})s, %(timezone)s)), %(timezone)s)"
-
-        if self.interval_annotation == "toStartOfWeek":
-            return f"toStartOfWeek(toDateTime(%({date_clause})s, %(timezone)s), 0)"
-
-        return clause
+            return f"toDateTime({datetime_expr}, %(timezone)s)"
+
+    @classmethod
+    def _truncate_normalized_datetime(cls, normalized_datetime_expr: str, trunc_func: str) -> str:
```

Comment:
This is the key function - ALL datetimes in queries should be wrapped with this, so that we don't forget about timezone conversion